### ISSUE TITLE
Should we name component parts based on the potential icon used or what we're trying to portray to the user

### ISSUE THREAD

**Issue by gregwhitworth** [WICG/open-ui#55]
I'm forking from the other issue as I think there were two seperate issues in one. This one and whether they should be referred to as glyphs/icons.

@levithomason I think this could be interesting, though I do wonder how semantics will play into this. Several teams across Microsoft use a pencil for "edit"; however, from a longevity standpoint, I think that might be a mistake to standardize on. Teams at MSFT who don't have edit states may also use that specific icon/glyph to signify "pencil" or something else. 

A better example may be that in Edge, a star signifies something like a "favorite". In the Store, that same glyph might be used to signify "rating". The icons are semantic for each team and their specific application. I'm curious how we could standardize considering how often this might happen. "Star" would convey the visual of the icons and could offer alignment for both teams, but it might not end up in the table we create simply because teams are naming the glyph for their semantic scenario.

Mind you, I don't have a great solution or alternative proposal - but I'm wondering how the matrix works with the above considered? Thoughts?

_Originally posted by @chrisdholt in https://github.com/WICG/open-ui/issues/27#issuecomment-558843244_

**Comment by gregwhitworth** [comment]
To put it succinctly, using @chrisdholt example - would you prefer for us to standardize on an icon being named: *star* or *favorite*?

I personally prefer to utilize the action or hint that is being conveyed as there are many paradigms and scenarios where the same icon may overlap and because we're desiring these to be blueprints for different platforms to build their controls/components from I don't think we should require that "edit" be represented by a "pencil" when they could prefer to solely use the word "Edit" or some other type of graphic.

**Proposed Resolution:** When creating a new control or component that has a part which may contain a graphic representation for an action or hint to the user, name it as the action or hint and not the assumed graphic representation.

**Comment by LJWatson** [comment]
Another benefit to this proposal is that it frames the icon in a way that is good for accessibility, where the intent or purpose of the image is the important piece of information, not whether it looks like a pencil or a tree or whatever.

**Comment by jouni** [comment]
I think this is the same issue as with colors, and there are two layers here. 

For colors, the base layer, has “blue”, “red“, “green”, etc. The second layer has “primary”, “error”, “success”, etc.

For icons, the base layer has “star”, “pencil”, “arrow right”, etc. The second layer has “favorite”, “edit”, “previous”, etc.

Then we have a mapping between the two on the design system level.

**Comment by gregwhitworth** [comment]
@jouni correct, but when naming the parts of the anatomy, say for a dropdown menu, we shouldn't standardize that the part that holds the icon as arrow as that doesn't give any indication as to what it does for AT and a design system may prefer to use a chevron or a + sign or any other icon that makes sense for what it does. I'm not saying I have a perfect name for that part yet but I think I side with naming them of the behavior or state change they plan to initiate. Thoughts?

**Comment by LJWatson** [comment]
I agree with jouni that there are layers of information for each thing, but think it'd make the system more usable if we handled them in order of specificity.

For example, we'll likely have N things that represent "Next". Of these some will be arrows and others will be chevrons. Of each of those there will be a multitude of colours, styles and other characteristics.

**Comment by jouni** [comment]
Maybe I misunderstood the purpose of this issue. I think I got stuck on the term “icon”. Now I suppose it was only used to describe the context, and not specifically talk about naming icons, but rather naming component parts that are usually represented using icons.

I agree that when naming parts of components, the parts should be named based on behavior rather than presentation.

**Comment by gregwhitworth** [comment]
@jouni awesome - yep that was the intent. That said there have been discussions about normalizing the names of the icons. Since my primary focus however is on the platform anatomies I probably won't put time into that. However, there is value for design systems to leverage the same names so if that is something you're passionate about I know we'd welcome the contribution/research.

**Comment by tomByrer** [comment]
I vote for using existing standards, eg in Unicode/Emoji it is **star**
Don't want to confuse folks. 
https://emojipedia.org/star/

Is this related to #27 ?

**Comment by gregwhitworth** [comment]
@tomByrer That is for the actual icon and I don't disagree with you on that as I have no desire to re-create the wheel there. The discussion for this thread, which I think we've landed on is that the container that the icon will be within, should not be named star. Because start could represent the star in the galaxies, a movie star, or a rating. The parts should be named on the action being envoked.

#27 is truly trying to determine, if we reference in spec terminology a graphic representation should we utilize icon or glyph. Doing the research across the web icon is the most commonly used term. So they're discussing similar pieces but separate issues entirely.

**Comment by gregwhitworth** [comment]
I've edited the title as using the word icon implies I'm talking about trying to standardize icon names which is not the goal of this issue. I also will be digging deeper into `<select>` sub parts this week and will have proposal on that shortly which will help inform the path moving forward as well.

**Comment by levithomason** [comment]
**TL;DR**

Seems there is consensus on naming according to behavior/usage/intent opposed to literal name describing the shape itself.  Agreed here 👍 

**Longer story**

IMHO it would be ideal to name icon slots by their usage or intent.  This allows folks to use any icon they would like without invalidating the name.  Example, an "expand-icon" (hypothetical) will always "expand" the component (such as a tree).  The actual shape of the glyph or graphic in that slot could be a "chevron-right / chevron-down" or "plus-sign / minus-sign" (hypothetical) and still not violate the fact that it is the "icon representing the expansion of the tree item."

If we name these by their actual name ("chevron-down"), then I would think that would also necessitate disallowing changing them to another glyph.  I would also propose that they must be able to changed else folks will build their own.

This is also inline with one of Open UI's current naming principals:

>### Names should be void of design information
>
>**Why?** When designs change names should not become invalid. Names should stand the test of time. Usually, names based on intent or usage are more robust and stand this test. Consider a component that displays mock content while waiting for the actual content to load. It might be more ideally named Placeholder opposed to Shimmer, even though most implementations appears to flash or shimmer. When the design changes and these components no longer shimmer, the name will become invalid and not make sense.

We also faced this issue over the last couple years in Stardust / Fluent UI since we allowed the icon set to be swapped out as part of the "theme" definition.  However, some components internally rely on certain icons being available (i.e. a Menu Item's "indicator" icon which points in the direction of a submenu).

**Comment by gregwhitworth** [comment]
Agreed. I'm going to resolve this issue, thank you all for your feedback.

> Resolved: Seems there is consensus on naming according to behavior/usage/intent opposed to literal name describing the shape itself.
